### PR TITLE
 Support for hierarchical account #73

### PR DESCRIPTION
**Pull request by wwjbatista**

### CODE REVIEW

File: app/controllers/kaui/accounts_controller.rb [hunk 1]
```diff
@@ -107,6 +111,8 @@ def show
     wait(fetch_payment_methods)
     @payment_methods = wait(fetch_payment_methods_with_details).map { |pm_f| pm_f.execute }.map { |pm_f| wait(pm_f) }.reject { |pm| pm.nil? }
     @available_tags = wait(fetch_available_tags)
+    @children = wait(fetch_children)
```
Comment (sbrossie) [inline app/controllers/kaui/accounts_controller.rb]: I am not super familiar with promises. Question: When will the call to fetch all the children accounts be executed? Is that when the user clicks on the `Children(N)` link?

Comment (wwjbatista) [reply app/controllers/kaui/accounts_controller.rb]: The promise here is only to check if the account has children, . When the user clicks the `Children(N)` link appear the list screen, and then via DataTable Ajax call it's fetches the children, but it is not done by the promise.

**Comment by sbrossie** [comment]
👍

File: app/views/kaui/accounts/index.html.erb [hunk 2]
```diff
@@ -42,12 +43,26 @@ $(document).ready(function() {
     "pageLength": <%= @limit %>,
     "displayStart": <%= @offset %>,
     <% unless @ordering.blank? %>
-        "order": [[ 0, "<%= @ordering %>" ]],
+        "order": [[ 1, "<%= @ordering %>" ]],
```
Comment (pierre) [inline app/views/kaui/accounts/index.html.erb]: Why this change? It's not consistent with other screens anymore.

File: app/views/kaui/accounts/index.html.erb [hunk 3]
```diff
@@ -42,12 +43,26 @@ $(document).ready(function() {
...
+    "rowCallback": function( row, data, index ) {
+      if ( data[0] == 1 ) {
+        $('td:eq(0)', row).html( '<span class="label label-info account-child-label">Child</span>' );
+      }else{
+        $('td:eq(0)', row).html('')
+      }
+    },
+    "columnDefs": [
+    { "orderable": false, "targets": 0, "createdCell": function (td, cellData, rowData, row, col) {
+        $(td).css('padding-left', '20px');
```
Comment (pierre) [inline app/views/kaui/accounts/index.html.erb]: Nit: might break responsiveness. Probably not worth fixing it right now though, as most screens would probably benefit from mobile testing again.

File: app/views/kaui/accounts/index.html.erb (hunk 2 above)
Comment (wwjbatista) [reply app/views/kaui/accounts/index.html.erb]: The change was because the first column was not the name anymore but a tag if the account is a child, and it will cause the UI to show an arrow when the first column is not orderable. I could restore it as it was, and mark the account as a child with an icon (http://fontawesome.io/icon/child/) after the name.

Comment (pierre) [reply app/views/kaui/accounts/index.html.erb]: Oh, you are right. My bad, I misunderstood the API. I thought this changed desc to asc or vice versa.

:shipit:
